InventoryBuilder.py

```diff
@@ -134,44 +161,65 @@ def query_inventory_permanent(self):
                 logger.debug(f'Withdrawing current iteration: {self.iteration}')
                 self.iterated_inventory.pop(str(self.iteration))
             self.iteration += 1
-            logger.info(f'Inventory relaxing before going to work again')
+            logger.info(f'Inventory relaxing before going to work again in {self.sleep}s')
             time.sleep(int(self.sleep))
 
-    def query_vrops(self, vrops):
+    def query_vrops(self, vrops, vrops_short_name):
         logger.info(f'Querying {vrops}')
         token = Vrops.get_token(target=vrops)
         if not token:
+            logger.warning(f'retrying connection to {vrops} in next iteration {self.iteration + 1}')
             return False
         self.target_tokens[vrops] = token
 
-        logger.info(f'##############################################')
-        logger.info(f'##########  Collecting resources... ##########')
-        logger.info(f'##############################################')
+        logger.info(f'##########  Collecting resources {vrops_short_name}... ##########')
 
         vcenter = self.create_resource_objects(vrops, token)
+        if not vcenter:
+            return False
         self.vcenter_dict[vrops] = vcenter
         return True
 
-    def create_resource_objects(self, vrops, token):
-        for adapter in Vrops.get_adapter(target=vrops, token=token):
-            logger.debug(f'Collecting vcenter: {adapter["name"]}')
-            vcenter = Vcenter(target=vrops, token=token, name=adapter['name'], uuid=adapter['uuid'])
-            vcenter.add_datacenter()
-            for dc_object in vcenter.datacenter:
-                logger.debug(f'Collecting datacenter: {dc_object.name}')
-                dc_object.add_cluster()
-                for cl_object in dc_object.clusters:
-                    logger.debug(f'Collecting cluster: {cl_object.name}')
-                    cl_object.add_host()
-                    for hs_object in cl_object.hosts:
-                        logger.debug(f'Collecting host: {hs_object.name}')
-                        hs_object.add_datastore()
-                        for ds_object in hs_object.datastores:
-                            logger.debug(f'Collecting datastore: {ds_object.name}')
-                        hs_object.add_vm()
-                        for vm_object in hs_object.vms:
-                            logger.debug(f'Collecting VM: {vm_object.name}')
-            return vcenter
+    def create_resource_objects(self, target, token):
+        vrops = Vrops()
+
+        name, uuid = Vrops.get_adapter(target, token)
+        if not name:
+            return False
+        logger.debug(f'Collecting vcenter: {name}')
+
+        datacenter = Vrops.get_resources(vrops, target, token, [uuid], resourcekinds=["Datacenter"])
+        vccluster = Vrops.get_resources(vrops, target, token, [dc.get('uuid') for dc in datacenter],
```

Comment:
shouldn't we go with cluster here instead?

Reply:
Nope. We use the parent uuids to fetch the resources.

Reply:
i meant variable naming `vcccluster`

Reply:
suggestion pls